### PR TITLE
CLN: remove LegacyFoo from io.pytables

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -180,8 +180,6 @@ class DuplicateWarning(Warning):
 # storer class map
 _STORER_MAP = {
     "Series": "LegacySeriesFixed",
-    "DataFrame": "LegacyFrameFixed",
```
Comment (jreback) [inline pandas/io/pytables.py]: i think Series as well?

**Review by jreback — CHANGES_REQUESTED** [review]
i would add a note to what’s new as well

**Comment by jbrockmendel** [comment]
> i would add a notet o what’s new as well

ok.  saying what?  its not actually clear to me what these unused classed _did_ that we are removing support for

**Comment by jreback** [comment]
just say that we removed legacy HDf5 support which was for pandas files written in version xxx (have to see if can figure when)

it’s  likely pretty old though like 0.11 or something

**Comment by jbrockmendel** [comment]
added a note, couldn't be too specific; looking through the blame i didnt see anything except "legacy", and the last relevant change was from 2013 with no attached PR

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**